### ISSUE TITLE
[BUG] AppService List ConnectionString Fails

### ISSUE THREAD

**Issue by ChenTanyi** [Azure/azure-sdk-for-net#38950]
### Library name and version

PackageVersion Include="Azure.ResourceManager.AppService" Version="1.0.2"

### Describe the bug

The `GetConnectionStringsAsync` throw exception after http 200

```
The requested operation requires an element of type 'Object', but the target element has type 'Array'.. Exception: System.InvalidOperationException: The requested operation requires an element of type 'Object', but the target element has type 'Array'.
   at Azure.ResourceManager.AppService.Models.ConnectionStringDictionary.DeserializeConnectionStringDictionary(JsonElement element)
   at Azure.ResourceManager.AppService.WebAppsRestOperations.ListConnectionStringsAsync(String subscriptionId, String resourceGroupName, String name, CancellationToken cancellationToken)
   at Azure.ResourceManager.AppService.WebSiteResource.GetConnectionStringsAsync(CancellationToken cancellationToken)
```

### Expected behavior

The function should return connection string successfully

### Actual behavior

See Describe

### Reproduction Steps

Can reproduce as it is from production log. We only have the request message. Maybe need to contact with service team about the response format.
{"requestMethod":"POST","requestUri":"https://management.azure.com/subscriptions/***/resourceGroups/***/providers/Microsoft.Web/sites/***/config/connectionstrings/list?api-version=2021-02-01","duration":444.3982,"correlationId":"f6de0155-17eb-45b0-9068-86328bc3190f","requestContentLength":0,"responseCode":200,"contentLength":807}

### Environment

Docker mcr.microsoft.com/dotnet/runtime:6.0

**Comment by mcgallan** [comment]
@ChenTanyi The issue you mentioned has been fixed in the latest unreleased version 1.1.0-beta.5. Thank you for your feedback. By the way, next time when you find a bug, we hope you can provide us with the code where the problem was found along with the error information. This will help us better carry out our work. Thank you again for your feedback.

**Comment by ChenTanyi** [comment]
Hi @mcgallan, thanks for the information! We will try to upgrade the SDK and tracking for error message.

As for code, this bug is throwed inside the SDK and it is just a "GET" request. So I don't think the code detail help. (That is why I only post the call stack and correlationId. For our code the exception is throwed at the first GET we send, the data is not POSTed by ourselves.)
